src/main/java/uk/gov/hmcts/reform/sscs/helper/PdfHelper.java

```diff
@@ -0,0 +1,125 @@
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDPage;
+import org.apache.pdfbox.pdmodel.PDPageContentStream;
+import org.apache.pdfbox.pdmodel.common.PDRectangle;
+import org.apache.pdfbox.util.Matrix;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class PdfHelper {
+
+    public boolean isDocumentWithinSize(PDDocument document, PDRectangle size) {
+
+        for (PDPage page: document.getPages()) {
+            float pageHeight = page.getMediaBox().getHeight();
+            float pageWidth = page.getMediaBox().getWidth();
+
+            if (pageHeight > pageWidth) {
+                if (pageHeight > size.getHeight() || pageWidth > size.getWidth()) {
+                    return false;
+                }
+            } else {
+                if (pageWidth > size.getHeight() || pageHeight > size.getWidth()) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    public BigDecimal calculateScalingFactor(PDDocument document, PDRectangle size) {
+
+        BigDecimal maxHeightScaling = new BigDecimal(0);
+        BigDecimal maxWidthScaling = new BigDecimal(0);
+        float heightOverage;
+        float widthOverage;
+
+        for (PDPage page: document.getPages()) {
+            float pageHeight = page.getMediaBox().getHeight();
+            float pageWidth = page.getMediaBox().getWidth();
+            float sizeHeight;
+            float sizeWidth;
+
+            if (pageHeight > pageWidth) {
+                sizeHeight = size.getHeight();
+                sizeWidth = size.getWidth();
+            } else {
+                sizeHeight = size.getWidth();
+                sizeWidth = size.getHeight();
+            }
+
+            log.debug("A4 height limit = " + sizeHeight);
+            log.debug("A4 width limit = " + sizeWidth);
+
+            log.debug("Page Height = " + pageHeight);
+            log.debug("Page Width = " + pageWidth);
+
+            heightOverage = pageHeight - sizeHeight;
+            log.debug("height overage = " + heightOverage);
+            widthOverage = pageWidth - sizeWidth;
+            log.debug("width overage = " + widthOverage);
+
+            maxHeightScaling = maxScaleFactor(heightOverage, pageHeight, maxHeightScaling);
+            log.debug("max height scaling = " + maxHeightScaling);
+            maxWidthScaling = maxScaleFactor(widthOverage, pageWidth, maxWidthScaling);
+            log.debug("max width scaling = " + maxWidthScaling);
+        }
+
+        BigDecimal scalingFactor = maxHeightScaling.compareTo(maxWidthScaling) > 0 ? maxHeightScaling : maxWidthScaling;
+
+        return new BigDecimal(1).subtract(scalingFactor).setScale(4, RoundingMode.HALF_EVEN);
+    }
+
+    public BigDecimal maxScaleFactor(float overage, float pageAxisDimension, BigDecimal maxScaleFactor) {
```

Comment:
I'm not sure why these methods are public, since they require expert knowledge to use them in combination. 

Is it possible to expose a public method like  Optional<PDDocument> resizeIfRequired(PDDocument originalDocument) ? 
There is no need to ask the size since, I presume it will always be A4.

Reply:
@ryandac  I've Made maxScaleFactor private private, calculateScalingFactor remaining public as useful both from verifying the maths in testing and general info about relative doc size.

Added 2 new helper methods for non-expert users:

 scaleToA4(PDDocument document) 
 scaleToPageSize(PDDocument document, PDRectangle size)
 
 
 Removed non-neccessary saves from Tests, shouldn't of been there!